src/Symfony/Component/Validator/Tests/Mapping/ClassMetadataTest.php

```diff
@@ -308,6 +309,19 @@ public function testGetPropertyMetadataReturnsEmptyArrayWithoutConfiguredMetadat
     {
         $this->assertCount(0, $this->metadata->getPropertyMetadata('foo'), '->getPropertyMetadata() returns an empty collection if no metadata is configured for the given property');
     }
+
+    public function testClassMetadataMembersAreUniqueByPropertyAndByClassAfterMergingConstraints()
+    {
+        $metadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $metadata->addGetterConstraint('data', new ConstraintA());
+
+        $otherMetadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $otherMetadata->addGetterConstraint('data', new ConstraintB());
+
+        $metadata->mergeConstraints($otherMetadata);
+
+        $this->assertCount(1, $metadata->getPropertyMetadata('data'));
```

Comment:
Indeed, the constraint doesn't matter. What matters in this test is that it's the same metadata class (`self::INTERFACE_A_CLASS`) and the same getter "property" (`data`). However, I agree that the constraint should maybe be the same so it's easier to understand.